lib/services/achievements-service.ts

```
@@ -0,0 +1,395 @@
+          break;
+
+        case 'saving_streak':
+          // This would require more complex logic to track saving streaks
+          // For now, we'll use a simple implementation
+          currentProgress = userStats.savingsTransactions;
+          shouldUnlock = currentProgress >= achievement.requirementValue;
+          break;
+
+        case 'budget_adherence':
+          // Check if user has been within budget limits
+          const budgetAdherence = await this.calculateBudgetAdherence(userId);
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)